server/src/main/java/org/opensearch/plugins/PluginInfo.java

```diff
@@ -81,17 +82,19 @@
      * @param opensearchVersion     the version of OpenSearch the plugin was built for
      * @param javaVersion           the version of Java the plugin was built with
      * @param classname             the entry point to the plugin
+     * @param customFolderName      the custom folder name for the plugin
      * @param extendedPlugins       other plugins this plugin extends through SPI
      * @param hasNativeController   whether or not the plugin has a native controller
      */
     public PluginInfo(String name, String description, String version, Version opensearchVersion, String javaVersion,
-                      String classname, List<String> extendedPlugins, boolean hasNativeController) {
+                      String classname, String customFolderName, List<String> extendedPlugins, boolean hasNativeController) {
```

Comment:
Is it possible to keep the original constructor and add a new constructor to support `customFolderName` ? This change caused AD test case failed. Not a big problem, but we'd better make the change backward compatible.

Reply:
Yes, we should have done that. Let's not break the interfaces like these :( 

/cc: @VachaShah 

@ylwu-amzn if you can PR this that'd be much appreciated

Reply:
@dblock  busy with AD plugin feature development. Will open an issue to track this.

Reply:
Created an issue: https://github.com/opensearch-project/OpenSearch/issues/1190